homeassistant/components/hlk_sw16/config_flow.py

```diff
@@ -0,0 +1,90 @@
+"""Config flow for HLK-SW16."""
+import asyncio
+
+from hlk_sw16 import create_hlk_sw16_connection
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_HOST, CONF_PORT
+from homeassistant.core import HomeAssistant
+
+from .const import (
+    CONNECTION_TIMEOUT,
+    DEFAULT_KEEP_ALIVE_INTERVAL,
+    DEFAULT_PORT,
+    DEFAULT_RECONNECT_INTERVAL,
+    DOMAIN,
+)
+from .errors import AlreadyConfigured, CannotConnect, NameExists
+
+DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_HOST): str,
+        vol.Optional(CONF_PORT, default=DEFAULT_PORT): vol.Coerce(int),
+    }
+)
+
+
+async def validate_input(hass: HomeAssistant, user_input):
+    """Validate the user input allows us to connect."""
+    for entry in hass.config_entries.async_entries(DOMAIN):
+        if (
+            entry.data[CONF_HOST] == user_input[CONF_HOST]
+            and entry.data[CONF_PORT] == user_input[CONF_PORT]
+        ):
+            raise AlreadyConfigured
+
+    client_aw = create_hlk_sw16_connection(
+        host=user_input[CONF_HOST],
+        port=user_input[CONF_PORT],
+        loop=hass.loop,
+        timeout=CONNECTION_TIMEOUT,
+        reconnect_interval=DEFAULT_RECONNECT_INTERVAL,
+        keep_alive_interval=DEFAULT_KEEP_ALIVE_INTERVAL,
+    )
+    try:
+        client = await asyncio.wait_for(client_aw, timeout=CONNECTION_TIMEOUT)
+    except asyncio.TimeoutError:
+        raise CannotConnect
+    else:
+        try:
+
+            def disconnect_callback():
+                if client.in_transaction:
+                    client.active_transaction.set_exception(CannotConnect)
+
+            client.disconnect_callback = disconnect_callback
+            await client.status()
+        except CannotConnect:
+            client.disconnect_callback = None
+            client.stop()
+            raise CannotConnect
+        else:
+            client.disconnect_callback = None
+            client.stop()
```

Comment:
Can we delete the top `else:` and save some indentation here ?

Reply:
removed